pkg/api/rest/rest.go

```diff
@@ -64,6 +64,14 @@ type Lister interface {
 	List(ctx api.Context, options *unversioned.ListOptions) (runtime.Object, error)
 }
 
+// Exporter is an object that knows how to strip a RESTful resource for export
+type Exporter interface {
+	// Export an object.  Fields that are not user specified (e.g. Status, ObjectMeta.ResourceVersion) are stripped out
+	// Modifies the object in place.  If 'exact' is true, fields that are specific to the cluster (e.g. namespace) are
```

Comment:
Minor nit... the comment doesn't apply anymore since you are returning the stripped object, not modifying it.